src/v2/models/AppState.js

```diff
@@ -328,6 +328,24 @@ export default Model.extend({
     const isCurrentAuthenticatorEmail = this.get('currentAuthenticatorEnrollment')?.key === AUTHENTICATOR_KEY.EMAIL;
 
     return isSameExceptMessages && isChallengeAuthenticator && isCurrentAuthenticatorEmail;
-  }
+  },
 
+  /**
+   * It will remove empty optional attributes from the model for attributes rendered as dropdown and text box.
```

Comment:
It will remove empty optional attributes from the view model of current view state.... `